src/parser.ts

```diff
@@ -84,6 +84,46 @@ function parseNonLiteral(
 ) {
   log('blue', 'parser', schema, '<-' + typeOfSchema(schema), processed.has(schema) ? '(FROM CACHE)' : '')
 
+  function parseUnionType(params: JSONSchema[]) {
```

Comment:
Not sure about putting this inside the function, but it seemed like the easiest way to get all of the parameters needed without making another function with basically the same signature as `parseNonLiteral`